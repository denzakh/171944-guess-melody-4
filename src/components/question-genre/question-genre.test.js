import React from "react";
import renderer from "react-test-renderer";
import QuestionGenre from "./question-genre.jsx";

const question = {
  type: `genre`,
  genre: `rock`,
  question: `Выберите инди-рок треки`,
  answers: [
    {
      genre: `rock`,
      src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Scott_Holmes/Inspiring__Upbeat_Music/Scott_Holmes_-_04_-_Upbeat_Party.mp3`
    },
    {
      genre: `blues`,
      src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Checkie_Brown/hey/Checkie_Brown_-_09_-_Mary_Roose_CB_36.mp3`
    },
    {
      genre: `jazz`,
      src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/Unheard_Music_Concepts/The_Lasso_of_Time/Unheard_Music_Concepts_-_24_-_Cosmic_Relevance.mp3`
    },
    {
      genre: `rock`,
      src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/West_Cortez_Records/David_Hilowitz/Gradual_Sunrise/David_Hilowitz_-_Gradual_Sunrise.mp3`
    }
  ]
};

describe(`QuestionGenre component`, ()=>{
  it(`should render QuestionGenre component markup`, () => {
    const tree = renderer.create(
        <QuestionGenre questionData={question} onAnswer={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
