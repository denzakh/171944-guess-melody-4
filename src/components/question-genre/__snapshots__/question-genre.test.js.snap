// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionGenre component should render QuestionGenre component markup 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx={390}
        cy={390}
        r={370}
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите инди-рок треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Scott_Holmes/Inspiring__Upbeat_Music/Scott_Holmes_-_04_-_Upbeat_Party.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-0"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Checkie_Brown/hey/Checkie_Brown_-_09_-_Mary_Roose_CB_36.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-1"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/Unheard_Music_Concepts/The_Lasso_of_Time/Unheard_Music_Concepts_-_24_-_Cosmic_Relevance.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-2"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-2"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://files.freemusicarchive.org/storage-freemusicarchive-org/music/West_Cortez_Records/David_Hilowitz/Gradual_Sunrise/David_Hilowitz_-_Gradual_Sunrise.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-3"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-3"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
